tools/shared.py

```diff
@@ -1212,25 +1212,35 @@ def make_jscall(sig):
   def make_dynCall(sig):
     # Optimize dynCall accesses in the case when not building with dynamic
     # linking enabled.
-    if not Settings.MAIN_MODULE and not Settings.SIDE_MODULE:
-      return 'dynCall_' + sig
-    else:
+    if Settings.MAIN_MODULE or Settings.SIDE_MODULE:
       return 'Module["dynCall_' + sig + '"]'
+    else:
+      return 'dynCall_' + sig
 
   @staticmethod
   def make_invoke(sig, named=True):
     legal_sig = JS.legalize_sig(sig) # TODO: do this in extcall, jscall?
     args = ','.join(['a' + str(i) for i in range(1, len(legal_sig))])
-    args = 'index' + (',' if args else '') + args
     ret = 'return ' if sig[0] != 'v' else ''
-    body = '%s%s(%s);' % (ret, JS.make_dynCall(sig), args)
+    args = 'index' + (',' if args else '') + args
+    if Settings.WASM_BIGINT or 'j' not in sig:
+      new_args = ['a' + str(i) for i in range(1, len(legal_sig))]
+      for i in range(len(sig) - 1):
+        if sig[i+1] == 'j':
+          new_args[i] = 'BigInt(%s)' % new_args[i]
```

Comment:
I don't get what this does. It converts the input into a BigInt, I think, but shouldn't we assume it already is a BigInt? (otherwise how can a user pass a value > 2^53?)

Reply:
Somehow `numbers` are reaching here.  

When i64 values flow out of WebAssembly and they less than 2^53 does the BitInt conversion still happen?  I would guess maybe not since that would explain the need to turn them into BitInt here.

Reply:
According to `https://webassembly.github.io/spec/js-api/#tojsvalue` it looks like we should always be getting BitInt when calling into JS so I need to investigate further.